src/tree/taffy_tree/tree.rs

```diff
@@ -57,22 +78,53 @@ impl Default for Taffy {
     }
 }
 
-/// Iterator that wraps a slice of nodes, lazily converting them to u64
-pub struct TaffyChildIter<'a>(core::slice::Iter<'a, NodeId>);
-impl<'a> Iterator for TaffyChildIter<'a> {
+/// Iterator over the Vec in a RefMut<'a, Vec<NodeId>>
+pub struct RefCellVecIter<'a> {
+    /// The items to iterate over
+    children: RefMut<'a, Vec<NodeId>>,
+    /// The next index to return
+    index: usize,
+}
+impl<'a> Iterator for RefCellVecIter<'a> {
     type Item = NodeId;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.0.next().copied()
+        let item = self.children.get(self.index).copied();
+        self.index += 1;
+        item
+    }
+}
+
+/// Iterates over children, checking the Display type of the node
+/// If the node is `Display::Contents`, then we recurse it's children, else we simply push the `NodeId` into the list
+fn find_children_recursive(tree: &Taffy, node: NodeId, out: &mut Vec<NodeId>) {
+    for child_id in tree.children[node.into()].iter() {
+        let child_key: DefaultKey = (*child_id).into();
+        let display = tree.nodes[child_key].style.display;
+        match display {
+            Display::Contents => find_children_recursive(tree, *child_id, out),
+            _ => out.push(*child_id),
+        }
     }
 }
 
 impl LayoutTree for Taffy {
-    type ChildIter<'a> = TaffyChildIter<'a>;
+    type ChildIter<'a> = RefCellVecIter<'a>;
 
     #[inline(always)]
     fn children(&self, node: NodeId) -> Self::ChildIter<'_> {
-        TaffyChildIter(self.children[node.into()].iter())
+        let mut cache = self.node_children_cache.borrow_mut();
+
+        // If the cache key does not match the requested node_id, then recompute the children for
+        // the requested node and update the cache in-place.
+        if cache.node_id != node {
+            cache.node_id = node;
+            cache.children.clear();
```

Comment:
I wonder if this could cause memory "leaks" as the vector will only grow and never shrink...
I can't think of a practical example where this would happen though, so we can probably ignore it for now

Reply:
Now that this is stored in `TaffyView` instead of `Taffy` it no longer leaks :)